### PR TITLE
ramips: add support for HATLab GateBoard-One

### PR DESCRIPTION
**Pull request by zzzz0317**
Hardware
--------
SoC:   MediaTek MT7621AT
RAM:   512MB DDR3
FLASH: 512KB SPI-NOR for BIOS (U-Boot) and 128MB for system
POWER: DC 5.5 4~24V or PD (Unknown version)
ETH:   5x RJ45 1000M switch port for LAN
       1x SFP/RJ45 1000M Combo port for WAN
BTN:   1x Reset Button
USB:   1x USB-A (USB 3.0)
       1x USB-C (USB 2.0 / PD / USB ACM)
PCIe:  3x Mini-PCI Express
SIM:   3x SIM slot

Known Issues
------------
1. SFP port may not work properly.
2. USB and MMC boot are not supported. If you have an older version of GateBoard-One (before V2.2), you can try soldering a W25N01GVZEIG chip to the reserved pad (tested on V2.1). You may need to upgrade the stock U-Boot to the latest version.
3. "/sbin/fixup-mac-address" is unable to write a new MAC address to the factory partition.
4. The onboard 8563 watchdog is currently not supported.

Installation
------------
1. Rename "kernel.itb" to "hatlab_gateboard-one-kernel.itb".
2. Rename "squashfs-rootfs.img" to "hatlab_gateboard-one-squashfs-rootfs.img".
3. Copy the two files to a TF card or USB flash disk, ensuring that the filesystem is FAT32.
4. Connect the Type-C serial port and repeatedly hit enter until the autoboot stops.
5. Enter the command "run flash_ubi_usb" if you are using a USB device or "run flash_ubi_mmc" if you are using a TF card.
6. Wait for the flashing process to complete and for OpenWRT to start up.

### CODE REVIEW

File: target/linux/ramips/dts/mt7621_hatlab_gateboard-one.dts [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+	cpu_sensor: lm75@4f {
+		compatible = "national,lm75";
+		reg = <0x4f>;
+		#thermal-sensor-cells = <0>;
+	};
+};
+
+&factory {
+	compatible = "nvmem-cells";
```
Comment (neheb) [inline target/linux/ramips/dts/mt7621_hatlab_gateboard-one.dts:382]: Please convert to nvmem-layout

**Comment by aiamadeus** [comment]
You need to commit to the main branch first.

File: target/linux/ramips/dts/mt7621_hatlab_gateboard-one.dts [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+			nvmem-cell-names = "mac-address";
+			mac-address-increment = <3>;
+		};
+
+		port@3 {
+			status = "okay";
+			label = "eth3";
+			nvmem-cells = <&macaddr_factory_0>;
+			nvmem-cell-names = "mac-address";
+			mac-address-increment = <4>;
```
[the remark this answers is not in the document]
Comment (zzzz0317) [reply target/linux/ramips/dts/mt7621_hatlab_gateboard-one.dts:331]: Modify it like this commit?

https://github.com/openwrt/openwrt/commit/b2f1c6ed52af4fbf80988d86037dbc90d9a08e6f

**Comment by zzzz0317** [comment]
> You need to commit to the main branch first.

OK, I will close this pr and make a new one.

File: target/linux/ramips/dts/mt7621_hatlab_gateboard-one.dts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zzzz0317) [reply target/linux/ramips/dts/mt7621_hatlab_gateboard-one.dts:331]: I modified the code to look like the following, removing the "mac-address-increment", but the MAC address incrementing did not take effect. How should I modify it to use MAC address auto increment?

```
			factory: partition@78000 {
				label = "factory";
				reg = <0x78000 0x8000>;

				nvmem-layout {
					compatible = "fixed-layout";
					#address-cells = <1>;
					#size-cells = <1>;

					macaddr_factory_0: macaddr@0 {
						compatible = "mac-base";
						reg = <0x0 0x6>;
					};
				};
			};

...

		port@3 {
			status = "okay";
			label = "eth3";
			nvmem-cells = <&macaddr_factory_0 4>;
			nvmem-cell-names = "mac-address";
		};
```

[the remark this answers is not in the document]
Comment (neheb) [reply target/linux/ramips/dts/mt7621_hatlab_gateboard-one.dts:331]: You’re missing the cell count change

**Pull request closed without merging**